buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/UpdateIntellijVersions.kt

```diff
@@ -0,0 +1,104 @@
+package com.virtuslab.gitmachete.buildsrc
+
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.getMajorPart
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.toBuildNumber
+import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.versionIsNewerThan
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import org.jsoup.Jsoup
+import org.jsoup.select.Elements
+
+open class UpdateIntellijVersions : DefaultTask() {
+
+  @get:Internal
+  val intellijReleasesContents: Elements by lazy {
+    getLinksFromUrl("https://www.jetbrains.com/intellij-repository/releases/")
+  }
+
+  @get:Internal
+  val intellijSnapshotsContents: Elements by lazy {
+    getLinksFromUrl("https://www.jetbrains.com/intellij-repository/snapshots/")
+  }
+
+  private fun getLinksFromUrl(repositoryUrl: String): Elements {
+    return Jsoup.connect(repositoryUrl).get()
+      .select("a[href^=${repositoryUrl}com/jetbrains/intellij/idea/ideaIC/][href$=.pom]")
+  }
+
+  private fun findFirstMatchingVersionNewerThan(repoLinks: Elements, regex: Regex, thresholdVersion: String): String? {
+    for (link in repoLinks) {
+      val attr = link.attr("href")
+      val matchResult = regex.find(attr)
+
+      if (matchResult != null) {
+        val foundVersion = matchResult.value
+
+        if (foundVersion versionIsNewerThan thresholdVersion) {
+          return foundVersion
+        }
+      }
+    }
+    return null
+  }
+  private fun findReleaseNewerThan(version: String): String? {
+    return findMatchingVersionNewerThan(
+      intellijReleasesContents,
+      Regex("(?<=ideaIC-)(\\d+\\.)+\\d+(?=.pom)"),
+      version
+    )
+  }
+
+  private fun findLatestMinorOfVersion(versionNumber: String): String {
+    val major = getMajorPart(versionNumber)
+
+    return findMatchingVersionNewerThan(
+      intellijReleasesContents,
+      Regex("(?<=ideaIC-)$major(\\.\\d+)?(?=.pom)"),
+      major
+    ) ?: versionNumber
+  }
+
+  private fun findEapWithBuildNumberHigherThan(buildNumber: String): String? {
+    return findMatchingVersionNewerThan(
+      intellijSnapshotsContents,
+      Regex("(?<=ideaIC-)\\d+\\.\\d+\\.\\d+(?=-EAP-SNAPSHOT.pom)"),
+      buildNumber
+    )
+  }
+
+  @TaskAction
+  fun execute() {
+    val properties = IntellijVersionHelper.getProperties()
+    val latestStable = IntellijVersions.latestStable
+    val latestMinorsList = IntellijVersions.latestMinorsOfOldSupportedMajors
+      .map { findLatestMinorOfVersion(it) }
+
+    properties.setProperty("latestMinorsOfOldSupportedMajors", latestMinorsList.joinToString(separator = ","))
+
+    val newerStable = findReleaseNewerThan(latestStable)
+
+    if (newerStable != null) {
+      project.logger.lifecycle("latestStable is updated to $newerStable")
+      properties.setProperty("latestStable", newerStable)
+      properties.remove("eapOfLatestSupportedMajor")
```

Comment:
Hmmm one nuance: what if we've got:
```properties
latestStable=2022.2.1
eapOfLatestSupportedMajor=223.1234.56-EAP-SNAPSHOT
```

and then 2022.2.2 comes out? IIUC it'll be picked up as `newerStable`, and hence `eapOfLatestSupportedMajor` will be (incorrectly!) removed :/

Reply:
Also! I wouldn't **remove** the property key, just set it to empty string